src/alignment_processor.py

```diff
@@ -60,6 +64,120 @@ def reset_region(self, chr_id, start, end):
         self._set(chr_id, start, end)
 
 
+class AbstractAlignmentStorage:
+    COVERAGE_BIN = 512
+
+    def __init__(self):
+        self.coverage_dict = defaultdict(int)
+        self.current_bin_region_start = 4294967296 # infinity
+        self.current_bin_region_end = 0
+
+    def reset(self):
+        self.coverage_dict = defaultdict(int)
+        self.current_bin_region_start = math.inf
```

Comment:
I don't think you want to use `math.inf` here, because on line 151 this value goes into `range`.

Very large int is probably fine (as in line 72) or `None` and an explicit check when needed.

Reply:
Thanks! Seems like I just forgot to correct it as I did in the constructor.

Reply:
and in that case the `import math` is unneeded